tasks/localizationtasks.js

```diff
@@ -24,228 +26,142 @@ const iso639_3_to_2 = {
 	trk: 'tr'
 };
 
+const supportedLocLangs = [
+	'chs',
+	'cht',
+	'deu',
+	'esn',
+	'fra',
+	'ita',
+	'jpn',
+	'kor',
+	'ptb',
+	'rus',
+];
+
+// Method that extracts all l10n.t calls from the source files and returns the l10n JSON object.
+async function getL10nJson() {
+	const srcFiles = await fs.readdir('./src', {
+		recursive: true
+	});
+	const tsFiles = srcFiles.filter(f => f.endsWith('.ts') || f.endsWith('.tsx'));
 
-// converts a json object into a plain text json
-function convertDictionaryToJson(dict) {
-	return JSON.stringify(dict, null, '\t') + '\n';
-}
-
-// converts an xml file into a json object
-function convertXmlToDictionary(xmlInput, escapeChar = true) {
-	let xmlDom = new dom().parseFromString(xmlInput);
-	let transUnits = xmlDom.getElementsByTagName('trans-unit');
-	let dict = {};
-	for (var i = 0; i < transUnits.length; ++i) {
-		let unit = transUnits[i];
-
-		// Extract ID attribute
-		let id = unit.getAttribute('id');
-
-		// Extract source element if possible
-		let sourceElement = unit.getElementsByTagName('source');
-		let source = '';
-		if (sourceElement.length >= 1) {
-			source = escapeChars(sourceElement[0].textContent, escapeChar);
-		}
-
-		// Extract target element if possible
-		let targetElement = unit.getElementsByTagName('target');
-		let target = '';
-		if (targetElement.length >= 1) {
-			target = escapeChars(targetElement[0].textContent, escapeChar);
+	const fileContents = [];
+	for (const file of tsFiles) {
+		const content = await fs.readFile(path.resolve('./src', file), 'utf8');
+		if (content) {
+			fileContents.push(content);
 		}
-
-		// Return json with {id:{target,source}} format
-		dict[id] = { 'source': source, 'target': target };
 	}
-
-	return dict;
+	const result = await vscodel10n.getL10nJson(fileContents.map(f => {
```

Comment:
Do you know if this will pick up both of these?

```
l10n.t('hello')
```

and

```
const { t } = l10n;
...
t('hello');
```

Reply:
added this note in the readme.